presentation/src/main/java/com/architect/coders/mu8/characters/CharactersActivity.kt

```diff
@@ -50,22 +36,28 @@ class CharactersActivity : AppCompatActivity() {
             )
         }
 
-        adapter = CharactersAdapter(viewModel::onCharacterClicked)
-        recycler.layoutManager = LinearLayoutManager(this)
-        recycler.adapter = adapter
+        binding.viewmodel = viewModel
+        binding.lifecycleOwner = this
 
-        viewModel.model.observe(this, Observer(::updateUi))
-    }
-
-    private fun updateUi(model: CharactersUiModel) {
```

Comment:
Why did you leave everything inside the `onCreate` method? It looks big, please create smaller methods for specific initializations and instead call them here.